pkg/frontend/frontend_scheduler_worker.go

```diff
@@ -309,8 +315,10 @@ func (w *frontendSchedulerWorker) runOne(ctx context.Context, client schedulerpb
 	backoff := backoff.New(ctx, backoffConfig)
 	for backoff.Ongoing() {
 		if !attemptLoop() {
+			w.forwardedRequestsFailedStats.Add(1)
 			backoff.Wait()
 		} else {
+			w.forwardedRequestsFailedStats.Set(0)
 			backoff.Reset()
 		}
 	}
```

Comment:
It would be great if we could distinguish between failed retry attempts and failures, I guess (assuming we want to setup alerting). `backoff.Err()` indicates whether we have exhausted all the attempts.  

I'm not very sure about resetting the metric with `w.forwardedRequestsFailedStats.Set(0)`, as this is a shared object; how does it affect concurrently occurring failures?